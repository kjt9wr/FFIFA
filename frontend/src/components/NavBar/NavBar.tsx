import { useState } from "react";
import {
  Button,
  Collapse,
  DropdownItem,
  DropdownMenu,
  DropdownToggle,
  Nav,
  NavItem,
  NavLink,
  Navbar,
  NavbarToggler,
  UncontrolledDropdown,
} from "reactstrap";
import { useLogout } from "../../custom-hooks/useLogout";
import { useAuthContext } from "../../custom-hooks/useAuthContext";
const NavBar = () => {
  const { logout } = useLogout();
  const { user } = useAuthContext();

  const [isOpen, setIsOpen] = useState(false);

  const toggle = () => setIsOpen(!isOpen);
  const handleClick = () => {
    logout();
  };

  return (
    <div>
      <Navbar
        className="navbar navbar-dark bg-dark navbar-expand-lg"
        fixed="top"
      >
        <NavbarToggler onClick={toggle} />
        <Collapse isOpen={isOpen} navbar>
          <Nav className="mr-auto" navbar>
            <NavItem>
              <NavLink href="/"> FFIFA </NavLink>
            </NavItem>
            <UncontrolledDropdown nav inNavbar>
              <DropdownToggle nav caret>
                Rosters
              </DropdownToggle>
              <DropdownMenu left={"true"}>
                <DropdownItem href="/roster/Kevin"> Kevin </DropdownItem>
                <DropdownItem href="/roster/Luigi"> Luigi </DropdownItem>
                <DropdownItem href="/roster/Casey"> Casey </DropdownItem>
                <DropdownItem href="/roster/Patrick"> Patrick </DropdownItem>
                <DropdownItem href="/roster/Jeff"> Jeff </DropdownItem>
                <DropdownItem href="/roster/Alex"> Alex </DropdownItem>

                <DropdownItem divider />
                <DropdownItem href="/roster/Michael"> Michael </DropdownItem>
                <DropdownItem href="/roster/Nikos"> Nikos </DropdownItem>
                <DropdownItem href="/roster/Brent"> Brent </DropdownItem>
                <DropdownItem href="/roster/Christian">Christian</DropdownItem>
                <DropdownItem href="/roster/Justin"> Justin </DropdownItem>
                <DropdownItem href="/roster/Matt"> Matt </DropdownItem>
              </DropdownMenu>
            </UncontrolledDropdown>
            <UncontrolledDropdown nav inNavbar>
              <DropdownToggle nav caret>
                Trades
              </DropdownToggle>
              <DropdownMenu right>
                <DropdownItem href="/trade"> Trade Tracker </DropdownItem>
                <DropdownItem href="/tradepreview">Preview Trade</DropdownItem>
              </DropdownMenu>
            </UncontrolledDropdown>
            <UncontrolledDropdown nav inNavbar>
              <DropdownToggle nav caret>
                Prices
              </DropdownToggle>
              <DropdownMenu right>
                <DropdownItem href="/franchise"> Franchise Tag </DropdownItem>
                <DropdownItem href="/supermax">Supermax</DropdownItem>
                <DropdownItem href="/arbitration">Arbitration</DropdownItem>
              </DropdownMenu>
            </UncontrolledDropdown>
            <NavItem>
              <NavLink href="/cap">Cap Tracker</NavLink>
            </NavItem>
            <NavItem>
              <NavLink href="/fa">Free Agency</NavLink>
            </NavItem>
            <NavItem>
              <NavLink href="/draft">Draft Day</NavLink>
            </NavItem>
          </Nav>
          <Nav className="ms-auto" navbar>
            {!user ? (
              <NavItem>
                <NavLink href="/Login">Login</NavLink>
              </NavItem>
            ) : (
              <UncontrolledDropdown nav inNavbar>
                <DropdownToggle nav caret>
                  {user.username}
                </DropdownToggle>
                <DropdownMenu right>
                  <DropdownItem href="/admin">Admin</DropdownItem>
                  <DropdownItem onClick={handleClick}>Logout</DropdownItem>
                </DropdownMenu>
              </UncontrolledDropdown>
            )}
          </Nav>
        </Collapse>
      </Navbar>
      <br /> <br /> <br />
    </div>
  );
};

export default NavBar;
