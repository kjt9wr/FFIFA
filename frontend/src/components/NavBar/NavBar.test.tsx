import { render, screen } from "@testing-library/react";
import NavBar from "./NavBar";
import { AuthContextProvider } from "../../context/AuthContext";
import "./localStorageMock.ts"; // Import the localStorage mock file
import userEvent from "@testing-library/user-event";

const user = { username: "dude" };
describe("Nav Bar", () => {
  it("renders successfully when logged out", () => {
    const view = render(
      <AuthContextProvider>
        <NavBar />
      </AuthContextProvider>
    );

    expect(view).toMatchSnapshot();
  });

  it("renders successfully when logged in", async () => {
    localStorage.setItem("user", JSON.stringify(user));
    render(
      <AuthContextProvider>
        <NavBar />
      </AuthContextProvider>
    );

    expect(await screen.findByText("Cap Tracker")).toBeTruthy();
  });

  it("can logout", async () => {
    localStorage.setItem("user", JSON.stringify(user));
    render(
      <AuthContextProvider>
        <NavBar />
      </AuthContextProvider>
    );

    userEvent.click(await screen.findByText("dude"));
    userEvent.click(await screen.findByText("Logout"));

    expect(screen.queryByText("Cap Tracker")).toBeFalsy();
  });
});
