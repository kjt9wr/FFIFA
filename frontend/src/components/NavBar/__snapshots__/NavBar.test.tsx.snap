// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Bar renders successfully when logged out 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="navbar navbar-custom navbar-expand-lg navbar fixed-top"
        >
          <div
            class="container-fluid"
          >
            <button
              aria-label="Toggle navigation"
              class="navbar-toggler"
              type="button"
            >
              <span
                class="navbar-toggler-icon"
              />
            </button>
            <div
              class="collapse navbar-collapse"
            >
              <ul
                class="secondaryBlue mr-auto navbar-nav"
              >
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/"
                  >
                     FFIFA 
                  </a>
                </li>
              </ul>
              <ul
                class="ms-auto navbar-nav"
              >
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/Login"
                  >
                    Login
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </nav>
        <br />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="navbar navbar-custom navbar-expand-lg navbar fixed-top"
      >
        <div
          class="container-fluid"
        >
          <button
            aria-label="Toggle navigation"
            class="navbar-toggler"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="collapse navbar-collapse"
          >
            <ul
              class="secondaryBlue mr-auto navbar-nav"
            >
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="/"
                >
                   FFIFA 
                </a>
              </li>
            </ul>
            <ul
              class="ms-auto navbar-nav"
            >
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="/Login"
                >
                  Login
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
      <br />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
